src/plugins/score-lib-process/Process/Script/ScriptEditor.cpp

```diff
@@ -150,4 +167,114 @@ void MultiScriptDialog::clearError()
   m_error->clear();
 }
 
+void ScriptDialog::openInExternalEditor()
+{
+  QString editorPath = QSettings{}.value("Skin/DefaultEditor").toString();
+
+  if(editorPath.isEmpty())
+  {
+    QMessageBox::warning(
+        this, tr("Error"), tr("no 'Default editor' configured in score settings"));
+    return;
+  }
+
+  QString tempFile = QStandardPaths::writableLocation(QStandardPaths::TempLocation)
+                     + "/ossia_script_temp.js";
+  QFile file(tempFile);
+  if(!file.open(QIODevice::WriteOnly))
+  {
+    QMessageBox::warning(this, tr("Error"), tr("failed to create temporary file."));
+    return;
+  }
+
+  file.write(this->text().toUtf8());
+  file.close();
+
+  auto& w = score::FileWatch::instance();
+  m_fileHandle = std::make_shared<std::function<void()>>([this, tempFile]() {
+    QFile file(tempFile);
+    if(file.open(QIODevice::ReadOnly))
+    {
+      QString updatedContent = QString::fromUtf8(file.readAll());
+      file.close();
+
+      QMetaObject::invokeMethod(m_textedit, [this, updatedContent]() {
+        if(m_textedit)
+        {
+          m_textedit->setPlainText(updatedContent);
+        }
+      });
+    }
+  });
+  w.add(tempFile, m_fileHandle);
+
+  if(!QProcess::startDetached(editorPath, QStringList() << tempFile))
```

Comment:
in 2024 I'd recommend `{ tempFile }` instead of `QStringList() << tempFile` as it is closer to how it is done in most "non-Qt" C++